### ISSUE TITLE
Span-overflow planner raises Unsupported after exhausting 512 split candidates for large tensors at high core counts

### ISSUE THREAD

**Issue by neha-ibm** [torch-spyre/torch-spyre#3417]
### What happened?

When the aggregate tensor span across all cores exceeds the total hardware limit (cores × 256 MiB), the span-overflow planner tries up to 512 candidate splits of the available host dimensions. 
The planner exhausts all 512 candidates and raises Unsupported:

> Unsupported: Spyre backend does not support: Cannot auto-tile buf0:
> no combined split among host dims [1] makes all spans fit within
> 255.996 MB after trying at most 512 combinations.

### What did you expect to happen?

The planner should either:

1. Successfully find a valid tiling by considering multi-dimensional splits (splitting across both d0 and d1 simultaneously rather than only per-dimension candidates), or
2. Increase the candidate budget beyond 512 when the tensor has high-factor row counts that require more search depth.

### How can we reproduce it?

```


import gc
import os

import torch
import torch._inductor.codecache

from torch_spyre._inductor import config as spyre_config
_SHAPE  = (270000, 16384)
_DTYPE  = torch.bfloat16
_CORES  = 32


def _logit_temperature(x):
    return x / 16.0


def _setup():
    torch.manual_seed(0xAFFE)
    torch._dynamo.reset_code_caches()
    torch._inductor.codecache.FxGraphCache.clear()
    os.environ.pop("SENCORES", None)
    os.environ["TORCHINDUCTOR_FORCE_DISABLE_CACHES"] = "1"
    spyre_config.sencores = _CORES


def _teardown():
    gc.collect()
    torch._dynamo.reset()
    os.environ.pop("TORCHINDUCTOR_FORCE_DISABLE_CACHES", None)
    os.environ.pop("SENCORES", None)
    spyre_config.sencores = 32


def scenario_automatic():
    print("\n--- Scenario: span-overflow planner exhausts 512 candidates ---")
    print(f"  Shape={_SHAPE}, dtype={_DTYPE}, cores={_CORES}")
    print(f"  Total = {_SHAPE[0] * _SHAPE[1] * 2 / 1024**2:.1f} MiB across {_CORES} cores")
    print("  Expected: Unsupported -- no combined split found after 512 attempts")
    _setup()
    with spyre_config.patch({"ignore_span_overflow_hints": False}):
        try:
            x = torch.randn(*_SHAPE, dtype=_DTYPE).to("spyre")
            torch._dynamo.reset()
            torch.compile(_logit_temperature, dynamic=False)(x)
            print("  RESULT: No exception raised")
        except Exception as e:
            print(f"  RESULT: Exception caught (expected): {type(e).__name__}: {e}")
    _teardown()


if __name__ == "__main__":
    scenario_automatic()

```

### Any environmental details we need to know?

Device: x86
Pytorch version: 2.11.0
Python version: 3.12.13

### Anything else we need to know?

Tested on torch-spyre commit: a4abedf

The error message says "host dims [1]" — meaning only one dimension (d0=270000) is considered as a host dim for splitting. Extending the search to consider joint splits across multiple dimensions, or increasing the candidate budget for shapes with large highly-composite row counts, would resolve this.

### Relevant log output

```Shell
--- Scenario: span-overflow planner exhausts 512 candidates ---
  Shape          : (270000, 16384)
  dtype          : torch.bfloat16
  cores          : 32
  Total tensor   : 8437.5 MiB
  Aggregate limit: 32 cores x 256 MiB = 8192 MiB
  Overflow by    : 245.5 MiB
  Host dims      : only d0=270000 available to split
  Why it fails   : no factor of 270000 produces per-core span <= 256.0 MiB
                   planner exhausts 512 candidates and raises Unsupported
  Expected       : InductorError wrapping Unsupported (NOT a SIGABRT)

  RESULT: Exception caught (expected)
  Type   : InductorError
  Message:
    Unsupported: Spyre backend does not support: Cannot auto-tile buf0:
    no combined split among host dims [1] makes all spans fit within
    255.996 MB after trying at most 512 combinations.
```
